pkg/cmd/validate.go

```diff
@@ -187,9 +192,43 @@ func ValidateFile(filePath string, resolver *validatorfactory.ValidatorFactory)
 	if err != nil {
 		return fmt.Errorf("error reading file: %w", err)
 	}
+	var documents [][]byte
+	if utils.IsYaml(filePath) {
+		reader := utilyaml.NewYAMLReader(bufio.NewReader(bytes.NewBuffer(fileBytes)))
+		for {
+			document, err := reader.Read()
+			if err == io.EOF || len(document) == 0 {
+				break
+			} else if err != nil {
+				return err
+			}
+			onlyComments := true
+			for _, line := range strings.Split(string(document), "\n") {
+				if strings.TrimSpace(line) == "" {
+					continue
+				} else if !strings.HasPrefix(line, "#") {
+					onlyComments = false
+					break
+				}
+			}
+			if !onlyComments {
+				documents = append(documents, []byte(document))
+			}
```

Comment:
@apelisse @alexzielenski does this make sense ?

Reply:
I think that mirror's kubectl's behavior. Though I think if the list of documents at the end is empty there should be an error.

Reply:
On a per file basis ?

Reply:
on a per-invocation basis. Mirroring kubectl's behavior

`kubectl-validate ./emptyfile ./pod.yaml` is accepted and validates pod.yaml
`kubectl-validate ./emptyfile` is an error since there were no documents given to validate
`kubectl-validate ./emptyfile ./anotheremptyfile` is an error since there were no documents given to validate

Reply:
ok, this means we have to load all files first then check for potential error and then eventually start processing.
it changes the logic we have today where we process one file at a time.

can we log an issue and address this in another PR ?

Reply:
sgtm